src/CabanaPD_HeatTransfer.hpp

```diff
@@ -0,0 +1,101 @@
+/****************************************************************************
+ * Copyright (c) 2022-2023 by Oak Ridge National Laboratory                 *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of CabanaPD. CabanaPD is distributed under a           *
+ * BSD 3-clause license. For the licensing terms see the LICENSE file in    *
+ * the top-level directory.                                                 *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#ifndef HEATTRANSFER_H
+#define HEATTRANSFER_H
+
+#include <CabanaPD_Timer.hpp>
+
+namespace CabanaPD
+{
+
+// Peridynamic heat transfer with forward-Euler time integration.
+// Inherits only because this is a similar neighbor-based kernel.
+template <class ExecutionSpace, class ModelType>
+class HeatTransfer : public Force<ExecutionSpace, BaseForceModel>
+{
+  protected:
+    using base_type = Force<ExecutionSpace, BaseForceModel>;
+    using base_type::_half_neigh;
+    using base_type::_timer;
+    ModelType _model;
+
+  public:
+    HeatTransfer()
+        : base_type( false ){};
+
+    HeatTransfer( const bool half_neigh, const ModelType model )
+        : base_type( half_neigh )
+        , _model( model )
+    {
+    }
+
+    template <class TemperatureType, class PosType, class ParticleType,
+              class NeighListType, class ParallelType>
+    void
+    computeHeatTransferFull( TemperatureType& temp, const PosType& x,
+                             const PosType& u, const ParticleType& particles,
+                             const NeighListType& neigh_list, const int n_local,
+                             ParallelType& neigh_op_tag, const double dt )
+    {
+        _timer.start();
+
+        auto model = _model;
+        const auto vol = particles.sliceVolume();
+
+        auto temp_func = KOKKOS_LAMBDA( const int i, const int j )
+        {
+            double xi, r, s;
+            getDistance( x, u, i, j, xi, r, s );
+
+            const auto vol = particles.sliceVolume();
+            const auto rho = particles.sliceDensity();
+            const double k = model.thermal_coeff * ( 1 - r / model.delta );
+            const double coeff = k * dt / rho( i ) / model.cp;
+            temp( i ) += coeff * ( temp( j ) - temp( i ) ) / r / r * vol( j );
+        };
+
+        Kokkos::RangePolicy<ExecutionSpace> policy( 0, n_local );
+        Cabana::neighbor_parallel_for(
+            policy, temp_func, neigh_list, Cabana::FirstNeighborsTag(),
+            neigh_op_tag, "CabanaPD::HeatTransfer::computeFull" );
+
+        _timer.stop();
+    }
+};
+
+// Forces with bond breaking.
```

Comment:
Why do we have a comment with "Forces" here?

Reply:
Copy-paste